dubbo-metadata/dubbo-metadata-report-redis/src/test/java/org/apache/dubbo/metadata/store/redis/RedisMetadataReportTest.java

```diff
@@ -60,15 +60,15 @@ public void constructor(TestInfo testInfo) throws IOException {
         String methodName = testInfo.getTestMethod().get().getName();
         if ("testAuthRedisMetadata".equals(methodName) || ("testWrongAuthRedisMetadata".equals(methodName))) {
             String password = "チェリー";
-            RedisServerBuilder builder = RedisServer.builder().port(redisPort).setting("requirepass " + password);
+            RedisServerBuilder builder = RedisServer.newRedisServer().port(redisPort).setting("requirepass " + password);
             if (SystemUtils.IS_OS_WINDOWS) {
```

Comment:
Yuck. I'm going to add a `.settingIf(boolean, String)` so that you can use the builder as it was intended.

Reply:
@codemonstur Thanks for your help. I hope everything will be the same as you expected.

Reply:
@codemonstur I see that your update has been merged in `master` branch. However, I still cannot work well when I run `RedisMetadataReportTest`.

If we want to fix issue #7421 , we need to upgrade the `redis` from `2.8.19` to `6.0.10` or `6.2 RC2` (see [#8062](https://github.com/redis/redis/issues/8062), [#8195](https://github.com/redis/redis/issues/8195)) in [embedded-redis](https://github.com/codemonstur/embedded-redis)

We can get some more details from [redis releases 6.0.10](https://github.com/redis/redis/releases/tag/6.0.10) 

Bug fixes:

+ Fix setproctitle related crashes. ([#8150](https://github.com/redis/redis/pull/8150), [#8088](https://github.com/redis/redis/pull/8088))
**Caused various crashes on startup, mainly on Apple M1 chips or under instrumentation**

I really hope you can commit an new PR to solve this problem (#7421).

Maybe @martin-g can persuade @codemonstur to do it.

Reply:
I was going to suggest to create an issue at https://github.com/codemonstur/embedded-redis and discuss there but "Issues" are not enabled there, only "Pull Requests".

@pinxiong As I said earlier - many of us do not have access to Mac ARM64 hardware and it is impossible for us to improve for it. But @codemonstur added APIs to use custom ExecutableProvider that can run any binary you tell it to.

Reply:
@pinxiong the current approach that I think will work for you is as @martin-g described. I just enabled issues in https://github.com/codemonstur/embedded-redis. So if you guys want to discuss any it is now possible.

Reply:
@martin-g @codemonstur I have opened an issue [#6](https://github.com/codemonstur/embedded-redis/issues/6) where we can discuss.